File: app/src/main/java/io/duckduckgosearch/app/HistoryItem.java
package io.duckduckgosearch.app;

import java.util.Date;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class HistoryItem {

    @PrimaryKey
    @NonNull
    public String searchTerm;

    @ColumnInfo(name = "search_date")
    public Date searchDate;

    public String getSearchTerm() {
        return searchTerm;
    }

    public Date getSearchDate() {
        return searchDate;
    }

    public HistoryItem(@NonNull String searchTerm, Date searchDate) {
        this.searchTerm = searchTerm;
        this.searchDate = searchDate;
    }

}

File: app/src/main/java/io/duckduckgosearch/app/HistoryItem.kt
package io.duckduckgosearch.app

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity
class HistoryItem internal constructor(@field:PrimaryKey var searchTerm: String, @field:ColumnInfo(name = "search_date") var searchDate: Date)